CMakeLists.txt

```diff
@@ -0,0 +1,115 @@
+###############################################################################
+#
+# (c) Copyright IBM Corp. 2016
+#
+#  This program and the accompanying materials are made available
+#  under the terms of the Eclipse Public License v1.0 and
+#  Apache License v2.0 which accompanies this distribution.
+#
+#      The Eclipse Public License is available at
+#      http://www.eclipse.org/legal/epl-v10.html
+#
+#      The Apache License v2.0 is available at
+#      http://www.opensource.org/licenses/apache2.0.php
+#
+# Contributors:
+#    Multiple authors (IBM Corp.) - initial implementation and documentation
+###############################################################################
+
+cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
+
+set(OMR_VERSION_MAJOR 0)
+set(OMR_VERSION_MINOR 0)
+set(OMR_VERSION_PATCH 1)
+set(OMR_VERSION ${OMR_VERSION_MAJOR}.${OMR_VERSION_MINOR}.${OMR_VERSION_PATCH})
+
+project(omr VERSION ${OMR_VERSION} LANGUAGES CXX C)
+
+set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules" ${CMAKE_MODULE_PATH})
+
+###
+### Getting the glue directory
+###
+
+set(OMR_GLUE "./example/glue" CACHE PATH "The glue directory")
+
+# TODO: OMR_EXAMPLE flag
+# TODO: OMR_RTTI flag
+
+# TODO: Version things
+
+set(OMR_VERSION_STRING "<Unknown>" CACHE STRING "") # TODO: Set the version
+set(OMR_JIT_VERSION_STRING "<Unknown>" CACHE STRING "") # TODO: Set the version
+
+###
+### Versions and stuff
+###
+
+include(cmake/platform.cmake)
+include(cmake/config.cmake)
+
+enable_testing()
+
+#include current source dir and current bin dir automatically
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+function(VPathResolve inlist dirlist outlist)
+    set(output "")
+  foreach(currentFile ${${inlist}})
+    set(fileFound False)
+    #message("Searching for ${currentFile}")
```

Comment:
if this diagnostic output is useful, maybe it would be better to introduce an option to control the output rather than just commenting out message lines.